src/Lib.P2P/Cryptography/StretchedKey.cs

```diff
@@ -0,0 +1,142 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Text;
+using Org.BouncyCastle.Crypto.Macs;
+using Org.BouncyCastle.Crypto.Parameters;
+using Org.BouncyCastle.Security;
+
+namespace Lib.P2P.Cryptography
+{
+    /// <summary>
+    ///   Symmetric keys for SECIO.
+    /// </summary>
+    /// <remarks>
+    ///   Keys derived from a shared secret.
+    /// </remarks>
+    public class StretchedKey
+    {
+        /// <summary>
+        ///   The nonce.
+        /// </summary>
+        public byte[] Iv { get; set; }
+
+        /// <summary>
+        ///   The message authentication code.
+        /// </summary>
+        public byte[] MacKey { get; set; }
+
+        /// <summary>
+        ///   The encyption key.
+        /// </summary>
+        public byte[] CipherKey { get; set; }
+
+        /// <summary>
+        ///   Create two streched keys from the shared secret.
+        /// </summary>
+        /// <remarks>
+        ///   The is no spec for this.  Copied https://github.com/libp2p/go-libp2p-crypto/blob/0f79fbebcb64f746a636aba79ece0635ec5919e9/key.go#L183
+        /// </remarks>
+        public static void Generate(string cipherName,
+            string hashName,
+            byte[] secret,
+            out StretchedKey k1,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider refactoring this method in order to remove the need for this 'out' modifier.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)